packages/pg-transaction/src/index.test.ts

```diff
@@ -0,0 +1,150 @@
+import { describe, it, before, beforeEach, after } from 'node:test'
+import { strict as assert } from 'assert'
+import { Client, Pool } from 'pg'
+import { transaction } from './index.js'
+
+const withClient = async (cb: (client: Client) => Promise<void>): Promise<void> => {
+  const client = new Client()
+  await client.connect()
+  try {
+    await cb(client)
+  } finally {
+    await client.end()
+  }
+}
+
+describe('Transaction', () => {
+  before(async () => {
+    // Ensure the test table is created before running tests
+    await withClient(async (client) => {
+      await client.query('CREATE TABLE IF NOT EXISTS test_table (id SERIAL PRIMARY KEY, name TEXT)')
+    })
+  })
+
+  beforeEach(async () => {
+    await withClient(async (client) => {
+      await client.query('TRUNCATE test_table')
+    })
+  })
+
+  after(async () => {
+    // Clean up the test table after running tests
+    await withClient(async (client) => {
+      await client.query('DROP TABLE IF EXISTS test_table')
+    })
+  })
+
+  it('should create a client with an empty temp table', async () => {
+    await withClient(async (client) => {
+      const { rowCount } = await client.query('SELECT * FROM test_table')
+      assert.equal(rowCount, 0, 'Temp table should be empty on creation')
+    })
+  })
+
+  it('should auto-commit at end of callback', async () => {
+    await withClient(async (client) => {
+      await transaction(client, async (client) => {
+        await client.query('INSERT INTO test_table (name) VALUES ($1)', ['AutoCommit'])
+        // row should be visible within transaction
+        const { rows } = await client.query('SELECT * FROM test_table')
+        assert.equal(rows.length, 1, 'Row should be inserted within transaction')
+
+        // while inside this transaction, the changes are not visible outside
+        await withClient(async (innerClient) => {
+          const { rowCount } = await innerClient.query('SELECT * FROM test_table')
+          assert.equal(rowCount, 0, 'Temp table should still be empty inside transaction')
+        })
+      })
```

Comment:
The pool version tests this, and the test description makes it sound like it was intended.

```suggestion

        // now that the transaction is committed, the changes are visible outside
        await withClient(async (innerClient) => {
          const { rowCount } = await innerClient.query('SELECT * FROM test_table')
          assert.equal(rowCount, 1, 'Row should be inserted after transaction commits')
        })
      })
```

(Used `rowCount` to match the preceding call; other places use `rows.length` 🤷)